### PR TITLE
feat: can add existing point (map location) to your route

### PR DESCRIPTION
**Pull request by Kondziow**
No ogolnie chyba dziala. Sposob Oskara, czyli z poziomu szczegolow trasy mozemy dodac punkt do naszej trasy.

Jak macie chwile to potestujcie czy nie ma bugow zadnych

### CODE REVIEW

**Review by Janek1010 — APPROVED** [review]

**Comment by Drillllll** [comment]
Kondzio popraw responsywność pliska
image

**Review by Drillllll — CHANGES_REQUESTED** [review]
layout do poprawy

File: src/app/route-map-location/route-map-location.service.ts [hunk 1]
```diff
@@ -1,23 +1,49 @@
...
+    };
+
+    return this.checkIfIsAlreadyCreated(routeId, mapLocationId).pipe(
+      switchMap(isCreated => {
+        if (!isCreated) {
+          const url = `${backendUrl}/routeMapLocations`;
+          return this.http.post(url, newRouteMapLocation);
+        } else {
+          alert('This map Location is already in your route!')
+          return of(null); // Zwraca Observable z wartością `null`, gdy rekord już istnieje
```
Comment (Drillllll) [inline src/app/route-map-location/route-map-location.service.ts:30]: change the comment to eng

File: src/app/route/route-list/route-list.component.ts [hunk 2]
```diff
@@ -34,9 +35,13 @@ export class RouteListComponent {
   user = null;
   selectedRoute: Route;
 
+  @Input() addingPointToRoute: boolean = false;
+  @Input() addingPointToRouteId: string;
```
Comment (Drillllll) [inline src/app/route/route-list/route-list.component.ts:39]: I would change the name. It's vague. I don't know if it refers to ML id or route id. 
maby "pointIdToBeAdded(toRoute)" ?

File: src/app/route/route-list/route-list.component.ts [hunk 3]
```diff
@@ -165,13 +169,25 @@ export class RouteListComponent {
...
+    if (this.addingPointToRoute) {
+      this.addPointToRoute()
+    }
+  }
+
+  addPointToRoute() {
+    if (confirm("You are about to add point to " + this.selectedRoute.name + " route.")) {
+      this.routeMapLocationService.postRouteMapLocation(this.selectedRoute.id, this.addingPointToRouteId)
+        .subscribe( response => {
+          this.router.navigate(['../list'], {relativeTo: this.activatedRoute})
```
Comment (Drillllll) [inline src/app/route/route-list/route-list.component.ts:189]: investigate the redirect

**Review by Drillllll — CHANGES_REQUESTED** [review]

**Comment by Drillllll** [comment]
Additionally the code in list is becoming a spaghetti with every task. What do you think about the structure like that?

image

**Comment by Drillllll** [comment]
by stealingML I mean adding ML to my route. It's shorter this way

**Comment by Drillllll** [comment]
This isn't a change request, but just my personal opinion. Just want to ask what u think

**Comment by Kondziow** [comment]
It seems to be a good idea but I think it requires a new task. It can be complex.

**Review by Drillllll — APPROVED** [review]

**Pull request merged**